src/components/Layout/Header/NavBar/ChainMenu.tsx

```diff
@@ -0,0 +1,124 @@
+import { selectAssetById, selectFeatureFlags } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+const ChainMenuItem: React.FC<{
+  chainId: ChainId
+  onClick: (chainId: ChainId) => void
+  isConnected: boolean
+}> = ({ chainId, onClick, isConnected }) => {
+  const chainName = getChainAdapters().get(chainId)?.getDisplayName()
+  const { chainReference: evmChainId } = fromChainId(chainId)
+  const nativeAssetId = getChainAdapters().get(chainId)?.getFeeAssetId()
+  const nativeAsset = useAppSelector(state => selectAssetById(state, nativeAssetId ?? ''))
+
+  const connectedIconColor = useColorModeValue('green.500', 'green.200')
+  const connectedChainBgColor = useColorModeValue('blackAlpha.100', 'whiteAlpha.50')
+
+  if (!nativeAsset) return null
+
+  return (
+    <MenuItem
+      icon={<AssetIcon src={nativeAsset.icon ?? ''} width='6' height='auto' />}
+      backgroundColor={isConnected ? connectedChainBgColor : undefined}
+      onClick={() => onClick(evmChainId)}
+      borderRadius='lg'
+    >
+      <Flex justifyContent={'space-between'}>
+        <Text>{chainName}</Text>
+        <Box>{isConnected && <CircleIcon color={connectedIconColor} w={2} />}</Box>
+      </Flex>
+    </MenuItem>
+  )
+}
+export const ChainMenu = () => {
+  const { state } = useWallet()
+  const [evmChainId, setEvmChainId] = useState<string | null>(null)
+  const featureFlags = useAppSelector(selectFeatureFlags)
+
+  const supportedEvmChainIds = useMemo(
+    () =>
+      Array.from(getChainAdapters().keys()).filter(
+        chainId => fromChainId(chainId).chainNamespace === CHAIN_NAMESPACE.Ethereum,
+      ),
+    // We want to explicitly react on featureFlags to get a new reference here
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [featureFlags],
+  )
+
+  useEffect(() => {
+    ;(async () => {
+      const chainId = await (state.wallet as any)?.ethGetChainId?.()
+      if (chainId) setEvmChainId(bnOrZero(chainId).toString())
+    })()
+  }, [state])
+
+  const connectedChainId = useMemo(
+    () => supportedEvmChainIds.find(chainId => fromChainId(chainId).chainReference === evmChainId),
+    [evmChainId, supportedEvmChainIds],
+  )
+
+  const handleChainClick = async (chainId: ChainId) => {
+    try {
+      await (state.wallet as any).ethSwitchChain?.(Number(chainId))
+      setEvmChainId(chainId)
+    } catch (e) {
+      // TODO: Handle me after https://github.com/shapeshift/hdwallet/pull/551 is published
+    }
+  }
+
+  const currentChainNativeAssetId = useMemo(
+    () =>
+      getChainAdapters()
+        .get(connectedChainId ?? '')
+        ?.getFeeAssetId(),
+    [connectedChainId],
+  )
+  const currentChainNativeAsset = useAppSelector(state =>
+    selectAssetById(state, currentChainNativeAssetId ?? ''),
+  )
+
+  if (!state.wallet || !evmChainId || !currentChainNativeAsset) return null
+  if (!supportsEthSwitchChain(state.wallet)) return null
+
+  return (
```

Comment:
how do we feel about only showing this menu if `supportedEvmChainIds.length > 1`? currently this just shows a useless button

Reply:
Ser I can see you tackled this in d4dc7ac11b1a1393b6dcbcce4b5723b40aa3def3, marking this as resolved